lib/elliptic/curve/edwards.js

```diff
@@ -61,7 +61,37 @@ EdwardsCurve.prototype.pointFromX = function pointFromX(odd, x) {
   if (odd && !isOdd || !odd && isOdd)
     y = y.redNeg();
 
-  return this.point(x, y, curve.one);
+  return this.point(x, y, this.one);
+};
+
+EdwardsCurve.prototype.pointFromY = function pointFromY(odd, y) {
+  var curve = this;
+
+  y = new bn(y, 16);
+  if (!y.red)
+    y = y.toRed(this.red);
+
+  // x^2 = (y^2 - 1) / (d y^2 + 1)
+  var y2 = y.redSqr();
+  var lhs = y2.redSub(curve.one);
+  var rhs = y2.redMul(curve.d).redAdd(curve.one);
+  var x2 = lhs.redMul(rhs.redInvm());
+
+  if (x2.cmpn(0) === 0) {
+    if (odd)
+      throw new Error('invalid point');
+    else
+      return curve.point(new bn(0).toRed(curve.red), y);
+  }
+
+  var x = x2.redSqrt();
+  if (x.redSqr().redSub(x2).cmpn(0) !== 0)
+    throw new Error('invalid point');
+
+  if (x.isOdd() !== odd)
+    x = x.redNeg();
+
+  return curve.point(x, y, curve.one);
```

Comment:
I just copied the curve.one stuff and odd as first arg as a nod to the
existing method. Will fix em both?